src/prefect/tasks.py

```diff
@@ -1634,7 +1634,43 @@ def task(
     refresh_cache: Optional[bool] = None,
     on_completion: Optional[list[StateHookCallable]] = None,
     on_failure: Optional[list[StateHookCallable]] = None,
-    retry_condition_fn: Optional[Callable[[Task[P, Any], TaskRun, State], bool]] = None,
+    retry_condition_fn: Literal[None] = None,
```

Comment:
This is the new thing that matches first when it is None and doesn't interact with returned `P` and `R`.